File: backend/src/models/SREPORT.ts
import { Model, DataTypes } from 'sequelize';
import db from '../config/database';

class SREPORT extends Model {
    public id!: any;
    public name!: any;
    public note!: any;
    public status!: any;
    public usermodifiedid!: any;
    public timemodified!: any;
    public timecreated!: any;
    public sortorder!: any;
    public usercreatedid!: any;
    public parentid!: any;
    public parentdir!: any;
    public itemtype!: any;
    public autoid!: any;
    public simageid!: any;
    public image32!: any;
    public filterconfig!: any;
    public verticallayout!: any;
    public runcount!: any;
    public masterview!: any;
    public detailview!: any;
    public hasdetail!: any;
    public filterontop!: any;
    public hasdetail2!: any;
    public detail2view!: any;
    public viewinreportmg!: any;
    public sql!: any;
    public params!: any;
    public lasttemplateid!: any;
    public sqlmode!: any;
    public reminder!: any;
    public format!: any;
    public thuongdung!: any;
    public code!: any;
    public colconfig!: any;
    public rawmode!: any;
    public classname!: any;
    public servercode!: any;
    public clientcode!: any;
}

SREPORT.init({
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true, autoIncrement: true,
    },
    NAME: {
        type: DataTypes.STRING,
        
    },
    NOTE: {
        type: DataTypes.STRING,
        
    },
    STATUS: {
        type: DataTypes.BOOLEAN,
        
    },
    USERMODIFIEDID: {
        type: DataTypes.INTEGER,
        
    },
    TIMEMODIFIED: {
        type: DataTypes.DATE,
        
    },
    TIMECREATED: {
        type: DataTypes.DATE,
        
    },
    SORTORDER: {
        type: DataTypes.INTEGER,
        
    },
    USERCREATEDID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTDIR: {
        type: DataTypes.STRING,
        
    },
    ITEMTYPE: {
        type: DataTypes.STRING,
        
    },
    AUTOID: {
        type: DataTypes.INTEGER,
        
    },
    SIMAGEID: {
        type: DataTypes.INTEGER,
        
    },
    IMAGE32: {
        type: DataTypes.BLOB,
        
    },
    FILTERCONFIG: {
        type: DataTypes.STRING,
        
    },
    VERTICALLAYOUT: {
        type: DataTypes.STRING,
        
    },
    RUNCOUNT: {
        type: DataTypes.STRING,
        
    },
    MASTERVIEW: {
        type: DataTypes.STRING,
        
    },
    DETAILVIEW: {
        type: DataTypes.STRING,
        
    },
    HASDETAIL: {
        type: DataTypes.BOOLEAN,
        
    },
    FILTERONTOP: {
        type: DataTypes.STRING,
        
    },
    HASDETAIL2: {
        type: DataTypes.BOOLEAN,
        
    },
    DETAIL2VIEW: {
        type: DataTypes.STRING,
        
    },
    VIEWINREPORTMG: {
        type: DataTypes.STRING,
        
    },
    SQL: {
        type: DataTypes.STRING,
        
    },
    PARAMS: {
        type: DataTypes.STRING,
        
    },
    LASTTEMPLATEID: {
        type: DataTypes.INTEGER,
        
    },
    SQLMODE: {
        type: DataTypes.STRING,
        
    },
    REMINDER: {
        type: DataTypes.STRING,
        
    },
    FORMAT: {
        type: DataTypes.STRING,
        
    },
    THUONGDUNG: {
        type: DataTypes.DECIMAL(18,2),
        
    },
    CODE: {
        type: DataTypes.STRING,
        
    },
    COLCONFIG: {
        type: DataTypes.STRING,
        
    },
    RAWMODE: {
        type: DataTypes.STRING,
        
    },
    CLASSNAME: {
        type: DataTypes.STRING,
        
    },
    SERVERCODE: {
        type: DataTypes.STRING,
        
    },
    CLIENTCODE: {
        type: DataTypes.STRING,
        
    },
}, {
    sequelize: db.sequelize,
    modelName: 'SREPORT',
    tableName: 'SREPORT',
    timestamps: false
});

export default SREPORT;

File: frontend/Models/SREPORT.cs
using System;

namespace QuanLyBar.Client.Models
{
    public class SREPORT
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public string Note { get; set; }
        public bool? Status { get; set; }
        public int? UsermodifiedId { get; set; }
        public DateTime? Timemodified { get; set; }
        public DateTime? Timecreated { get; set; }
        public int? Sortorder { get; set; }
        public int? UsercreatedId { get; set; }
        public int? ParentId { get; set; }
        public string Parentdir { get; set; }
        public string Itemtype { get; set; }
        public int? AutoId { get; set; }
        public int? SimageId { get; set; }
        public byte[] Image32 { get; set; }
        public string Filterconfig { get; set; }
        public string Verticallayout { get; set; }
        public string Runcount { get; set; }
        public string Masterview { get; set; }
        public string Detailview { get; set; }
        public bool? Hasdetail { get; set; }
        public string Filterontop { get; set; }
        public bool? Hasdetail2 { get; set; }
        public string Detail2view { get; set; }
        public string Viewinreportmg { get; set; }
        public string Sql { get; set; }
        public string Params { get; set; }
        public int? LasttemplateId { get; set; }
        public string Sqlmode { get; set; }
        public string Reminder { get; set; }
        public string Format { get; set; }
        public decimal? Thuongdung { get; set; }
        public string Code { get; set; }
        public string Colconfig { get; set; }
        public string Rawmode { get; set; }
        public string Classname { get; set; }
        public string Servercode { get; set; }
        public string Clientcode { get; set; }
    }
}
